File: typescript-agent/agent.ts
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";

const apiKey = process.env.COVASYN_API_KEY;
if (!apiKey) {
  console.error("Set COVASYN_API_KEY first.");
  process.exit(1);
}

const transport = new StdioClientTransport({
  command: "npx",
  args: ["-y", "@covasyn/mcp-client"],
  env: { ...process.env, COVASYN_API_KEY: apiKey },
});

const client = new Client({ name: "covasyn-example", version: "0.0.1" }, { capabilities: {} });
await client.connect(transport);

const tools = await client.listTools();
console.log(`Connected — ${tools.tools.length} tools`);

const result = await client.callTool({
  name: "covabasic_canonicalize",
  arguments: { smiles: "CC(=O)Oc1ccccc1C(=O)O" },
});
console.log("Aspirin canonical:", result.content);

await client.close();

File: python-agent/agent.py
"""Minimal example: spawn CovaSyn MCP client and invoke a tool."""
import asyncio
import os

from mcp import ClientSession, StdioServerParameters
from mcp.client.stdio import stdio_client


async def main() -> None:
    api_key = os.environ.get("COVASYN_API_KEY")
    if not api_key:
        raise SystemExit("Set COVASYN_API_KEY first.")

    server_params = StdioServerParameters(
        command="npx",
        args=["-y", "@covasyn/mcp-client"],
        env={"COVASYN_API_KEY": api_key},
    )

    async with stdio_client(server_params) as (read, write):
        async with ClientSession(read, write) as session:
            await session.initialize()

            tools = await session.list_tools()
            print(f"Connected to CovaSyn — {len(tools.tools)} tools available")

            result = await session.call_tool(
                "covabasic_canonicalize",
                arguments={"smiles": "CC(=O)Oc1ccccc1C(=O)O"},
            )
            print(f"Aspirin canonical SMILES: {result.content}")


if __name__ == "__main__":
    asyncio.run(main())
